client_plugin/drivers/shared/shared_driver.go

```diff
@@ -111,7 +233,118 @@ func (d *VolumeDriver) UnmountVolume(name string) error {
 
 // Create - create a volume.
 func (d *VolumeDriver) Create(r volume.Request) volume.Response {
-	log.Errorf("VolumeDriver Create to be implemented")
+	log.Infof("Entering VolumeDriver Create")
+	var msg string
+	var hnames []string
+	var entries []kvPair
+
+	// Ensure that the node running this command is a manager
+	info, err := d.dockerd.Info(context.Background())
+	if err != nil {
+		msg = fmt.Sprintf("Failed to get Info from docker client. Reason: %v", err)
+		log.Warningf(msg)
+		return volume.Response{Err: msg}
+	}
+	if info.Swarm.ControlAvailable == false {
+		msg = fmt.Sprintf("This node is not a swarm manager.")
+		msg += fmt.Sprintf(" Shared Volume creation is only possible from swarm manager nodes.")
+		log.Warningf(msg)
+		return volume.Response{Err: msg}
+	}
+
+	// Initialize volume metadata in KV store
+	hostname, err := os.Hostname()
+	if err != nil {
+		msg = fmt.Sprintf("Failed to fetch hostname during volume creation")
+		log.Warningf(msg)
+		return volume.Response{Err: msg}
+	}
+	hnames = append(hnames, hostname)
+	volRecord := VolumeMetadata{
+		Status:         volStateCreating,
+		GlobalRefcount: 0,
+		Port:           0,
+		ServiceName:    "Not started yet",
```

Comment:
My suggestion is, do not set all the fields below "Port" during initialization and add omitempty tag in the struct definition:

```
	Port           int       `json:"port,omitempty"`
 	ServiceName    string    `json:"serviceName,omitempty"`
 	Username       string    `json:"username,omitempty"`
 	Password       string    `json:"password,omitempty"`
 	ClientList     []string  `json:"clientList,omitempty"`
```

So they will be empty fields when they are not set.